config/initializers/rack_attack.rb

```diff
@@ -1,6 +1,34 @@
 # frozen_string_literal: true
-# No more that 500 requersts per minitue per ip
-limit = ENV.fetch('HYKU_ATTACK_RATE_LIMIT', 500).to_i
+
+# No more that 20 requests per minute per ip
+limit = ENV.fetch('HYKU_ATTACK_RATE_LIMIT', 20).to_i
 period = ENV.fetch('HYKU_ATTACK_RATE_PERIOD', 60).to_i
 throttle_off = ActiveModel::Type::Boolean.new.cast(ENV.fetch('HYKU_ATTACK_RATE_THROTTLE_OFF', Rails.env.development? || Rails.env.test?))
-Rack::Attack.throttle("requests by ip", limit: limit, period: period, &:ip) unless throttle_off
+
+unless throttle_off
+  Rack::Attack.throttle('throttle catalog requests by ip', limit: limit, period: period) do |req|
+    req.ip if req.path.starts_with?('/catalog')
```

Comment:
If behind a load balancer/proxy, `req.ip` is returning the load balancer IP, which is not what we want. We should explore using `req.get_header('HTTP_X_ORIGINAL_FORWARDED_FOR')` and falling back on `req.ip` if `HTTP_X_ORIGINAL_FORWARDED_FOR` is blank

Reply:
or we should figure out which deployments this is happening for and fix them ;-)

Reply:
Can you elaborate? I'm hesitant to merge this before addressing the issue; with a request limit of 20, and `req.ip` being the LBs, I think a busy site could effectively DoS itself for all users

Reply:
which apps have the request.ip as the LBs. those apps are broken and should be fixed. if you grab a list, I can help you solve them

Reply:
Hyku staging for sure, that's where I've been testing